Sources/Utility/Process.swift

```diff
@@ -0,0 +1,295 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import class Foundation.ProcessInfo
+
+import enum POSIX.SystemError
+import libc
+import Basic
+
+/// Process allows spawning new subprocesses and working with them.
+public final class Process {
+    /// Typealias for process id type.
+    public typealias ProcessID = pid_t
+
+    /// The arguments to execute.
+    public let arguments: [String]
+
+    /// The enviornment with which the process was executed.
+    public let environment: [String: String]
+
+    /// The return code of the process, available once process has finished execution.
+    public private(set) var returnCode: Int32 = 0
+
+    /// The process id of the spawned process, available after the process is launched.
+    public private(set) var processID = ProcessID()
+
+    /// The stream containing output of the process, data is available only after execution.
+    public private(set) var outputStream: OutputByteStream? = nil
+
+    /// If the subprocess has launched.
+    public private(set) var launched = false
+
+    /// If the subprocess received a signal.
+    public private(set) var signalled = false
```

Comment:
It doesn't seem to really be used internally, either (except for setting it to `true`).